src/eatery/common_eatery.py

```diff
@@ -101,6 +108,82 @@ def parse_events(event_list, event_date):
     new_events.append(new_event)
   return new_events
 
+def parse_expanded_menu(eatery):
+  """Parses the expanded menu of an eatery.
+
+  Returns a list of FoodCategoryTypes available in the external expandedItems resource
+
+  Args:
+    eatery (dict): A valid json segment from Cornell Dining that contains eatery information
+  """
+  eatery_slug = eatery['slug']
+  menus = requests.get(STATIC_EXPANDED_ITEMS_URL).json()
+  categories = []
+
+  for menu in menus['eateries']:
+    if menu['slug'] == eatery_slug:
+      for category in menu['categories']:
+        food_category = FoodCategoryType(
+          category=category['category'],
+          stations=parse_expanded_stations(category['stations'])
+        )
+        categories.append(food_category)
+
+  return categories
+
+def parse_expanded_stations(eatery_stations):
+  """Parses the expanded stations of an eatery.
+
+  Returns a list of DescriptiveFoodStationTypes available to a certain category of an eatery.
+
+  Args:
+    eatery_stations (dict): A valid json segment from the hard-coded menu items
+  """
+  stations = []
+  for station in eatery_stations:
+    expanded_station = DescriptiveFoodStationType(
+      category=station['station'],
+      items=parse_expanded_items(station['diningItems'])
+    )
+    stations.append(expanded_station)
+  return stations
+
+def parse_expanded_items(eatery_items):
+  """Parses the expanded items for each station.
+
+  Returns a list of DescriptiveFoodItemTypes for an eatery's station.
+
+  Args:
+    eatery_items (dict): A valid json segment from the hard-coded menu items
+  """
+  items = []
+  for item in eatery_items:
+    food_item = DescriptiveFoodItemType(
+      item=item['item'],
+      healthy=False,
+      options=parse_expanded_item_options(item['choices']),
+      price=item['price']
+    )
+    items.append(food_item)
+  return items
+
+def parse_expanded_item_options(item_options):
+  """Parses the options for an item.
+
+  Returns a list of DescriptiveFoodItemOptionTypes for an item.
+
+  Args:
+    item_options (dict): A valid json segment from the hard-coded menu items
+  """
+  option_categories = []
+  for option in item_options:
+    item_option = DescriptiveFoodItemOptionType(
+      label=option['label'],
+      options=option['options']
+    )
+    option_categories.append(item_option)
+  return option_categories
```

Comment:
Perhaps a more nifty way to write all of these functions, and more clear as well:

```python
option_to_category = lambda option: DescriptiveFoodItemOptionType(
  label=option['label'],
  options=option['options']
)
return map(option_category, item_options)
```

@cswenberg @meganle thoughts?

Reply:
I'm not familiar with python's `lambda` features but if it works, it works. @young-k

Reply:
I followed the conventions of a lot of the other parsing functions in the code, we could potentially have all of them use anonymous functions?